scripts/autonomous_cross_deduplicator_par.py

```diff
@@ -0,0 +1,151 @@
+                        help='the Jaccard similarity threshold (0.9).')
+    parser.add_argument('--processes', '-P', type=int, default=1,
+                        help='number of worker processes to use (max is the '
+                             'num of cores, default: 1). Note that in order '
+                             'to deduplicate documents, much memory might be '
+                             'needed, so it is a good idea to be conservative '
+                             'with the number of processes.')
+    parser.add_argument('--log-level', '-L', type=str, default='info',
+                        choices=['debug', 'info', 'warning',
+                                 'error', 'critical'],
+                        help='the logging level.')
+    args = parser.parse_args()
+    if not args.output_dir.is_dir():
+        parser.error('The directory for the batches must exist.')
+    if args.temp_dir:
+        if not args.temp_dir.is_dir():
+            parser.error('The temporary directory, if set, must exist.')
+    return args
+
+
+def has_minhash_content(directory: Path) -> bool:
+    """
+    Tells whether a given directory contains all the following files or not:
+    1.doc_ids, 1.files, 1.minhashes
+    """
+    return ((directory / '1.doc_ids').is_file()
+            and (directory / '1.files').is_file()
+            and (directory / '1.minhashes').is_file())
+
+
+def assemble_targets(input_dir: Path, output_dir: Path,
+                     from_dir: Path, upto_dir):
+    """
+    Returns a list of tuples. The tuples contain the parameters for the
+    deduplicate_other() method:
+    The input (minhash self) dir of the target
+    The list of dirs of older batches (minhash full)
+    The output (minhash full) dir for the target.
+
+    Ignores dirs which do not contain the required files
+    """
+    full_list = [dir.name for dir in sorted(input_dir.iterdir())
+                 if has_minhash_content(dir)]
+    if upto_dir:
+        try:
+            upto_i = full_list.index(upto_dir)
+        except ValueError:
+            logging.error('The upto-dir is not a valid batch')
+            raise
+        full_list = full_list[: upto_i+1]
+    if from_dir:
+        try:
+            from_i = full_list.index(from_dir)
+        except ValueError:
+            logging.error('The from-dir is not a valid batch')
+            raise
+        target_list = full_list[from_i:]
+    else:
+        from_i = 0
+        target_list = full_list
+    logging.debug(f'The list of targets is: {target_list} \n They start from '
+                  f'pos {from_i} of the relevant history: {full_list}')
+    pairings = []
+    for index, target in enumerate(target_list):
+        # TODO we do not support multiple minhash files per batch.
+        target_as_input = input_dir / target / '1'
+        target_as_output = output_dir / target
+        past = [output_dir / dir / '1' for dir in full_list[: from_i + index]]

```

Comment:
I wouldn't leave a space after the `:`. I think PEP-8 in this regard is not what it should be, because I think e.g. `[i:i + 1]` seems a bit strange, but flake8 definitely shouts if there is a space before, so for the sake of symmetry, let's use `:` without spaces on either side in this context.

Reply:
flake8 was fine with both versions.